src/features/rewards/categoryLikePicker/index.tsx

```diff
@@ -0,0 +1,113 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import {
+  StyledCategoryActionOptinLogo,
+  StyledCategoryActionOptoutLogo,
+  StyledCategoryActionOptinButton,
+  StyledCategoryActionOptoutButton,
+  StyledCategoryActionOptinFilledButton,
+  StyledCategoryActionOptoutFilledButton
+} from './style'
+import {
+  HeartLIcon,
+  HeartSIcon,
+  BlockLIcon,
+  BlockSIcon
+} from '../../../components/icons'
+import { Tooltip } from '..'
+import { getLocale } from '../../../helpers'
+
+interface State {
+  itemSelected: number
+}
+
+export interface Props {
+  id?: string
+  optAction: number
+  onOptIn?: () => void
+  onOptOut?: () => void
+}
+
+export default class ThumbLikePicker extends React.PureComponent<Props, State> {
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      itemSelected: props.optAction
+    }
+  }
+
+  componentWillReceiveProps (nextProps: Props) {
+    if ('optAction' in nextProps) {
+      this.setState({ itemSelected: nextProps.optAction })
+    }
+  }
+
+  render () {
+    return (
+      this.state.itemSelected === 1 ?
```

Comment:
maybe moving this into separate function so that it's easier to read it

Reply:
Donel